handler/proxy.go

```diff
@@ -611,6 +649,26 @@ func setHeaderFields(header http.Header, options OptionsMap) {
 	}
 }
 
+func collectQueryParams(ctx *hcl.EvalContext, attr *hcl.Attribute, dest map[string][]string) {
```

Comment:
This method seems to be obsolete at all. Additionally the error handling could be improved.

Please refer to `NewCtxOptions` which returns the required type already with proper hcl diagnostics handling.